src/passes/Asyncify.cpp

```diff
@@ -503,7 +508,12 @@ class ModuleAnalyzer {
       // Only the functions in the whitelist can change the state.
       for (auto& func : module.functions) {
         if (!func->imported()) {
-          map[func.get()].canChangeState = whitelist.count(func->name) > 0;
```

Comment:
This looks like the behavior might have changed - before we always assigned to `canChangeState`, but in this PR it becomes only if it we turn it true. Maybe this explains the test failure?

Reply:
Exactly. I didn't realize asyncify would apply auto-detection only to be overwritten by the whitelist. Perharps this can be optimized out, unless there's something left to do with imported functions.